src/components/tests/SourcesTree.spec.js

```diff
@@ -6,32 +6,617 @@ import React from "react";
 import { shallow } from "enzyme";
 import SourcesTree from "../../components/PrimaryPanes/SourcesTree";
 import * as I from "immutable";
+import { showMenu } from "devtools-contextmenu";
+import { copyToTheClipboard } from "../../utils/clipboard";
+import {
+  getShownSource,
+  getSelectedSource,
+  getDebuggeeUrl,
+  getExpandedState,
+  getProjectDirectoryRoot,
+  getSources
+} from "../../selectors";
+
+jest.mock("devtools-contextmenu", () => ({ showMenu: jest.fn() }));
+jest.mock("../../utils/clipboard", () => ({ copyToTheClipboard: jest.fn() }));
+jest.mock("../../selectors");
+
+const defaultSources = I.Map({
+  "server1.conn13.child1/39": createMockSource(
+    "server1.conn13.child1/39",
+    "http://mdn.com/one.js"
+  ),
+  "server1.conn13.child1/40": createMockSource(
+    "server1.conn13.child1/40",
+    "http://mdn.com/two.js"
+  ),
+  "server1.conn13.child1/41": createMockSource(
+    "server1.conn13.child1/41",
+    "http://mdn.com/three.js"
+  )
+});
+
+const singleMockSource = I.Map({
+  "server1.conn13.child1/41": createMockSource(
+    "server1.conn13.child1/41",
+    "http://mdn.com/three.js"
+  )
+});
+
+const singleMockItem = createMockItem(
+  "http://mdn.com/one.js",
+  "one.js",
+  I.Map({ id: "server1.conn13.child1/39" })
+);
+
+const blackboxedMockItem = createMockItem(
+  "http://mdn.com/blackboxed.js",
+  "blackboxed.js",
+  I.Map({ id: "server1.conn13.child1/59" })
+);
+
+const blackboxedMockSource = I.Map({
+  "server1.conn13.child1/59": createMockSource(
+    "server1.conn13.child1/59",
+    "http://mdn.com/blackboxed.js",
+    true
+  )
+});
+
+const mockRootDirectory = {
+  contents: [],
+  name: "root",
+  path: "root/"
+};
+
+const mockDirectory = {
+  contents: [],
+  name: "folder",
+  path: "folder/"
+};
+
+const mockEvent = {
+  preventDefault: jest.fn(),
+  stopPropagation: jest.fn()
+};
+
+let defaultComponent;
+let defaultProps;
+let defaultState;
```

Comment:
I did recommend changing `currState` to `defaultState`. component didn't need to be renamed to `defaultComponent` because it is mutated and you continue to use the same object. calling it default component makes it more confusing, as you expect it to change.